inc/os_linux.h

```diff
@@ -9,6 +9,7 @@
 //
 #define _GNU_SOURCE
 #define __USE_GNU
+#define __USE_MISC // needed for SOL_TCP in <netinet/tcp.h>
```

Comment:
I have just read http://man7.org/linux/man-pages/man7/feature_test_macros.7.html
which says that __USE_MISC and __USE_GNU should not be defined here.
It is defined in <features.h> if _BSD_SOURCE or _SVID_SOURCE is defined.
Furthermore a comment in features.h explains:
  

>  The `-ansi' switch to the GNU C compiler, and standards conformance
>    options such as `-std=c99', define __STRICT_ANSI__.  If none of
>    these are defined, or if _DEFAULT_SOURCE is defined, the default is
>    to have _SVID_SOURCE, _BSD_SOURCE, and _POSIX_SOURCE set to one and
>    _POSIX_C_SOURCE set to 200809L.  If more than one of these are
>    defined, they accumulate.  For example __STRICT_ANSI__,
>    _POSIX_SOURCE and _POSIX_C_SOURCE together give you ISO C, 1003.1,
>    and 1003.2, but nothing else.

So I have to modify the above.